test/local/mysql_tests.js

```diff
@@ -1,388 +1,369 @@
-              },
-              function(err) {
-                t.fail('Error when creating a stored procedure' + err)
-              }
-            )
-            .then(function() {
-              // monkey patch the DB so that we're doing what the other writes to stored procedures are doing
-              db.testStoredProcedure = function() {
-                var callProcedure = 'CALL testStoredProcedure()'
-                return this.write(callProcedure)
-              }
-              return db.testStoredProcedure()
-            })
-            .then(function() {
-              t.fail('The call to the stored prodcedure should have failed')
-            }, function(err) {
-              t.pass('The call to the stored procedure failed as expected')
-              t.equal(err.code, 500, 'error code is correct')
-              var possibleErrors = [
-                { msg: 'ER_BAD_NULL_ERROR', errno: 1048 },
-                { msg: 'ER_NO_DEFAULT_FOR_FIELD', errno: 1364 }
-              ]
-              var matchedError = false
-              possibleErrors.forEach(function(possibleErr) {
-                if (err.message === possibleErr.msg) {
-                  if (err.errno === possibleErr.errno ) {
-                    matchedError = true
-                  }
-                }
-              })
-              t.ok(matchedError, 'error message and errno are correct')
-            })
-        }
-      )
-
-      test(
-        'readMultiple with valid queries',
-        function (t) {
-          t.plan(9)
-          return db.readMultiple([
-            {
-              sql: 'SELECT * FROM accounts LIMIT ?',
-              params: 1
-            },
-            {
-              sql: 'SELECT COUNT(*) AS count FROM accounts WHERE createdAt < ? AND normalizedEmail LIKE ?',
-              params: [ Date.now(), '%@mozilla.com' ]
-            }
-          ])
+      var writer = function() {
+        ++callCount
+        return db.write(query, param)
           .then(
-            function(results) {
-              t.ok(Array.isArray(results), 'results array was returned')
-              t.equal(results.length, 2, 'results array contained two items')
-              t.ok(Array.isArray(results[0]), 'first result was an array')
-              t.ok(results[0].length <= 1, 'first result contained zero or one items')
-              t.ok(Array.isArray(results[1]), 'second result was an array')
-              t.equal(results[1].length, 1, 'second result contained one item')
-              t.equal(typeof results[1][0], 'object', 'second result item was object')
-              t.equal(Object.keys(results[1][0]).length, 1, 'second result item had one property')
-              t.ok(results[1][0].count >= 0, 'count property was non-negative number')
+            function(result) {
+              assert(false, 'this query should never succeed!')
+            },
+            function(err) {
+              assert(true, 'we got an error')
```

Comment:
This guy doesn't look like he's doing much. The others have `assert(err, ...)`, should we update him to be the same?